### DISCUSSION TITLE
Using Markdown inside of tabularx or commands

### DISCUSSION THREAD

**Discussion by FGUFA** [Witiko/markdown#148] [Q&A] [answered]
I'm a complete latex novice, so it might be that I'm missing something very obvious. I'm fiddling with trying to embed some Markdown snippets into a Latex document, and I keep running into obscure Latex errors trying to put Markdown into a tabularx table. And I get similar errors when trying to use Markdown inside of a command definition. Is this some known limitation or am I doing something very wrong?

Here a small example of what I'm trying to do:

```
\documentclass{article}

\usepackage{markdown}
\usepackage{tabularx}

\begin{document}

\begin{markdown}
This __works__ !
\end{markdown}

\begin{tabularx}{\linewidth}{|X|X|X|}
This works &
\begin{itemize}
\item This
\item works
\end{itemize} &
\begin{markdown}
This does *NOT* !
\end{markdown}
\end{tabularx}

\newcommand{\someTestCommand}{
\begin{markdown}
This *also* doesn't work!
\end{markdown}
}

\someTestCommand

\end{document}
```

And here are the sort of errors I'm getting with this code:

```
example.tex:21: Argument of ^^M has an extra }.
<inserted text> 
                \par 
l.21 \end{tabularx}
                   
Runaway argument?
 This does *NOT* ! \end {markdown} \endtabular 
example.tex:21: Paragraph ended before ^^M was complete.
<to be read again> 
                   \par 
l.21 \end{tabularx}
                   
example.tex:21: Extra }, or forgotten \endgroup.
<recently read> }
                 
l.21 \end{tabularx}
```

**Comment by Witiko** [comment, marked as answer]
@FGUFA `\begin{markdown} ... \end{markdown}` must appear at the top level of a document. It cannot be produced as a result of expansion (`\newcommand`) or inside environments that scan their content (`\begin{tabular}`, `\begin{frame}`). This is because `\begin{markdown}` must be able to scan its contents first before TeX gets to incorrectly tokenize it according to TeX's rules, throwing away what it perceives as comments after percent signs and all sorts of other nasty things.

If you save your markdown document into a separate file (such as `my-document.md`) and use the `\markdownInput{my-document.md}` command, this limitation no longer applies. Admittedly, this is less convenient.

**Reply by FGUFA** [reply]
Thanks! I was already suspecting something like this, but couldn't find a definitive statement on it. `\markdownInput` is working for me, and I think I can deal with the inconvenience :)
